### ISSUE TITLE
Call `LoseContext` to prevent `WARNING: Too many active WebGL contexts. Oldest context will be lost.`

### ISSUE THREAD

**Issue by Gykonik** [jacomyal/sigma.js#1386]
Similar to [#1271](https://github.com/jacomyal/sigma.js/issues/1271), I've ran into `WARNING: Too many active WebGL contexts. Oldest context will be lost.` because the context is not properly cleaned up even after calling `.clear()` and `.kill`.

What I've tried is the following (inside `ngOnDestroy`):
```
this._chartRenderer.clear();
this._chartRenderer.kill();
this._chartRenderer = null;
```
      
However, this did not solve the issue and I've come up with the following solution. However, the "type cheating" is quite hacky and I feel like this should be the default behaviour of a `.kill()` function (to properly clean up the WebGL contexts), shouldn't it?

```
this._chartRenderer.clear();
this._chartRenderer.kill();
Object.values((this._chartRenderer as any).webGLContexts as WebGLRenderingContext[]).forEach(
  context: WebGLRenderingContext) => {
    // Free up all the context
    context.getExtension("WEBGL_lose_context")?.loseContext();
  }
}
this._chartRenderer = null;
```

**Comment by jacomyal** [comment]
Interesting, thanks for this feedback. I'm a bit afraid of just pushing a fix, because of this comment from @Yomguithereal:
https://github.com/jacomyal/sigma.js/issues/1271#issuecomment-1278876667

Just so that I know what to target: Which version of sigma are you using?

**Comment by Gykonik** [comment]
Yea,h I've also read through this post. And tbh, I have no idea whether this has any implications or stuff. I can only say that it works for me and I have (a) no warnings in the console and (b) I guess it also frees up the resources (at least, the warnings with "too many WebGL Contexts disappeared).

But sure, this could have bad side-effects :D 

I'm using the newest version 3.0.0-alpha3

**Comment by jacomyal** [comment]
OK, good to know, I don't have to push any hot-fix to the v2.4 version, at least :)

@Yomguithereal Any take on that?

**Comment by Gykonik** [comment]
Thanks for the explanation! I'm not entirely sure whether this is related to angular or not. I already use the `DestroyRef` by putting it inside `ngOnDestroy` that gets automatically called when the component gets destroyed. And as expected, I also see the `"DESTROY"`-message every time I expect something to be destroyed. Nonetheless, it still doesn't work without the `loseContext` - with it, it works flawlessly :D

```  
public ngOnDestroy(): void {
    this.graph?.clear();

    console.log("DESTROY!");
    // Proper cleanup to properly delete WebGL context.
    if (this.renderer) {
      this.renderer.clear();
      this.renderer.kill();
      // Free up all the context (see https://github.com/jacomyal/sigma.js/issues/1386)
      // Object.values((this.renderer as any).webGLContexts as WebGLRenderingContext[]).forEach(
      //   (context: WebGLRenderingContext) => {
      //     context.getExtension("WEBGL_lose_context")?.loseContext();
      //   }
      // );
      this._renderer = null;
    }
}
  ```
